### ISSUE TITLE
Soap Call failing without getting any error

### ISSUE THREAD

**Issue by hvaidsain26** [circa10a/easy-soap-request#59]
I am making a soap call like this but only thing that gets printed is the log statement in catch block without any err object.
am I doing something wrong here.
try {
		const { response } = await soapRequest({
			url: url,
			headers: {
				"Content-Type": "text/xml;charset=UTF-8",
				soapAction: actionUrl,
			},
			xml: reqBody,
			timeout: 10000,
		});
		const { headers, body, statusCode } = response;
		console.log(headers);
		console.log(body);
		console.log(statusCode);
		
	} catch (err) {
		console.log("Error is Soap API call", err);
	}

**Comment by circa10a** [comment]
It seems the `response.data` object isn't getting populated by [axios](https://github.com/axios/axios) under the hood. Can you try the following code? It's just your code you have in the issue with the source of `easy-soap-request`, except it's returning the entire `error` object.

```js
function soapRequest(opts = {
    method: 'POST',
    url: '',
    headers: {},
    xml: '',
    timeout: 10000,
    proxy: {},
    maxBodyLength: Infinity,
    maxContentLength: Infinity,
    extraOpts: {},
}) {
    const {
        method,
        url,
        headers,
        xml,
        timeout,
        proxy,
        maxBodyLength,
        maxContentLength,
        extraOpts,
    } = opts;
    return new Promise((resolve, reject) => {
        axios({
            method: method || 'POST',
            url,
            headers,
            data: xml,
            timeout,
            proxy,
            maxBodyLength,
            maxContentLength,
            ...extraOpts,
        }).then((response) => {
            resolve({
                response: {
                    headers: response.headers,
                    body: response.data,
                    statusCode: response.status,
                },
            });
        }).catch((error) => {
            reject(error);
        });
    });
};

try {
    const {
        response
    } = await soapRequest({
        url: url,
        headers: {
            "Content-Type": "text/xml;charset=UTF-8",
            soapAction: actionUrl,
        },
        xml: reqBody,
        timeout: 10000,
    });
    const {
        headers,
        body,
        statusCode
    } = response;
    console.log(headers);
    console.log(body);
    console.log(statusCode);
} catch (err) {
    // Robbed from https://github.com/axios/axios#handling-errors
    if (error.response) {
        // The request was made and the server responded with a status code
        // that falls out of the range of 2xx
        console.log(error.response.data);
        console.log(error.response.status);
        console.log(error.response.headers);
    } else if (error.request) {
        // The request was made but no response was received
        // `error.request` is an instance of XMLHttpRequest in the browser and an instance of
        // http.ClientRequest in node.js
        console.log(error.request);
    } else {
        // Something happened in setting up the request that triggered an Error
        console.log('Error', error.message);
    }
}
```

**Comment by Killroyk2** [comment]
I'm having the exact same issue, although I was actually able to get a response on the first 2 attempts and when I tried to test it again it failed but without any error message.

**Comment by circa10a** [comment]
@Killroyk2 could you also try the function I posted above?

**Comment by circa10a** [comment]
I've pushed a fix for this in `5.0.0`. `easy-soap-request` now returns the entire error object from axios, making debugging more flexible.

[Link to documentation for the error object being returned](https://github.com/axios/axios#handling-errors)
